### PR TITLE
GPU fix

### PR DESCRIPTION
**Pull request by KirillZubov**

### CODE REVIEW

File: src/pinns_pde_solve.jl [hunk 1]
```diff
@@ -600,7 +600,7 @@ function get_numeric_derivative()
     derivative =
         (phi,u,x,εs,order,θ) ->
         begin
-            _epsilon = 1 / (2*cbrt(eps(eltype(θ))))
+            _epsilon = 1.f0 / (2*cbrt(eps(eltype(θ))))
```
Comment (ChrisRackauckas) [inline src/pinns_pde_solve.jl:603]:
```suggestion
            _epsilon = one(eltype(θ)) / (2*cbrt(eps(eltype(θ))))
```

File: src/pinns_pde_solve.jl [hunk 2]
```diff
@@ -631,7 +631,7 @@ function generate_random_points(points, bound)
            fill(Float32(b),(1,points))
        else
            lb, ub =  b[1], b[2]
-           lb .+ (ub .- lb) .* Float32.(rand(1,points))
+           Float32.(lb .+ (ub .- lb) .* (rand(1,points)))
```
Comment (ChrisRackauckas) [inline src/pinns_pde_solve.jl:634]: `Float32` should never be used. It should all be based off of input types.

**Comment by ChrisRackauckas** [comment]
I think something happened with your commit? It should be like `eltype(x)` and `convert.(eltype(x),...)` etc.

File: src/pinns_pde_solve.jl [hunk 3]
```diff
@@ -156,8 +156,8 @@ function transform_expression(ex,dict_indvars,dict_depvars, chain,initθ, strate
 end
 
 function get_ε(dim, der_num)
-    epsilon = cbrt(eps(Float32))
-    ε = zeros(Float32, dim)
+    epsilon = cbrt(eps(eltypeθ))
```
Comment (ChrisRackauckas) [inline src/pinns_pde_solve.jl:159]:
```suggestion
    epsilon = cbrt(eps(eltype(θ)))
```

Comment (KirillZubov) [reply src/pinns_pde_solve.jl:159]: we do not have access to θ from this method, and the method is executed before training when assembling the model, so there is no need to do dynamic determine the type of θ.

Comment (ChrisRackauckas) [reply src/pinns_pde_solve.jl:159]: got it

Comment (ChrisRackauckas) [reply src/pinns_pde_solve.jl:159]: it's never dynamic BTW if type-stable. It's hoisted to compile time.

Comment (KirillZubov) [reply src/pinns_pde_solve.jl:159]: yes, it is type-stable always,  and it is not necessary to define the type dynamically as we did before in the loss function(`DiffEqBase.parameterless_type(θ)`), it is enough to do it once during compilation.

Comment (ChrisRackauckas) [reply src/pinns_pde_solve.jl:159]: gotcha

**Comment by KirillZubov** [comment]
@ChrisRackauckas  can we calculate in Float64? or the chain still works with accuracy float32 and just convert with noise in the last figures?
 ```julia
 chain = FastChain(FastDense(1,20,Flux.σ),FastDense(20,1))
 initθ = Float64.(DiffEqFlux.initial_params(chain))
 chain(rand(1,3),initθ)
1×3 Matrix{Float64}:
 -0.254752  -0.241568  -0.276606
```

**Comment by ChrisRackauckas** [comment]
If the weight and the input are Float64 then the computational is at Float64 accuracy.

**Comment by KirillZubov** [comment]
OK then there needs research that and add tests with Float64

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**